### PR TITLE
Enable simulation runs API

### PR DESCRIPTION
**Pull request by polomani**
Users can enable the use of the simulator runs API by setting in the config.yaml:
```
connector:
     use-simulators-api: true
```

- test is disabled until we have 
1. persistence for the eventId on the simulation run object
2. runTime parameter that allows to choose time range of the simulation
3. a bit more filters (or sort) on the simulation runs list

### CODE REVIEW

File: Cognite.Simulator.Tests/UtilsTests/SimulationRunnerTest.cs [hunk 1]
```diff
@@ -231,6 +250,160 @@ public async Task TestSimulationRunnerBase()
...
+                // Run the simulation runner and verify that the event above was picked up for execution
+                using var linkedTokenSource2 = CancellationTokenSource.CreateLinkedTokenSource(source.Token);
+                var linkedToken2 = linkedTokenSource2.Token;
+                linkedTokenSource2.CancelAfter(TimeSpan.FromSeconds(15));
+                var taskList2 = new List<Task> { runner.Run(linkedToken2) };
+                await taskList2.RunAll(linkedTokenSource2).ConfigureAwait(false);
+
+                Assert.True(runner.MetadataInitialized);
+
+                // Uncomment when we have runTime param in place
```
Comment (polomani) [inline Cognite.Simulator.Tests/UtilsTests/SimulationRunnerTest.cs:337]: not runTime but validatioEndTime

Comment (polomani) [reply Cognite.Simulator.Tests/UtilsTests/SimulationRunnerTest.cs:337]: change aggregation type for stepInterpolation and in that case the test will succeed regardless of wether we have dps or not

File: Cognite.Simulator.Utils/SimulationRunnerBase.cs [hunk 2]
```diff
@@ -411,68 +581,98 @@ public async Task Run(CancellationToken token)
...
+            if (simEv.HasSimulationRun)
+            {
+                if (!simEv.Run.EventId.HasValue)
+                {
+                    _logger.LogDebug("Simulation run has no Event associated with it {Id}", simEv.Run.Id);
+                    return;
+                }
+                simEvent = await _cdfEvents.GetAsync(simEv.Run.EventId.Value, token);
             }
+            else
```
Comment (polomani) [inline Cognite.Simulator.Utils/SimulationRunnerBase.cs:671]: fix warning (for example docs on public methods)

**Review by lprovensi — APPROVED** [review]
Approved, as this was discussed during office hours

**Pull request merged**